### DISCUSSION TITLE
how to add push button to existing service i.e. CO2

### DISCUSSION THREAD

**Discussion by homonto** [HomeSpan/HomeSpan#756] [Q&A]
Hi, 
So I have created a device that has few services, Eve presents it this way:

<img width="608" alt="Screenshot 2024-01-29 at 18 48 06" src="https://github.com/HomeSpan/HomeSpan/assets/46562447/35040f5c-3e21-4f5b-8037-d2ea020c45df">

<img width="425" alt="Screenshot 2024-01-29 at 18 40 39" src="https://github.com/HomeSpan/HomeSpan/assets/46562447/11a21fe0-74d6-4dd0-bb10-aea6de97b1ac">

this one: `struct DEV_Identify : Service::AccessoryInformation`  has also the characteristic `Characteristic::Identify(); ` that presents a button on iphone called ID and I am using it to initiate firmware update (in Eve - I cannot find it in Home App):

<img width="293" alt="Screenshot 2024-01-29 at 18 49 10" src="https://github.com/HomeSpan/HomeSpan/assets/46562447/d5b0b2a1-10e1-46c5-9b8b-af7a7f007034">

My sensor CO2 has 3 characteristics: CarbonDioxideLevel, CarbonDioxidePeakLevel and CarbonDioxideDetected.

<img width="372" alt="Screenshot 2024-01-29 at 18 49 51" src="https://github.com/HomeSpan/HomeSpan/assets/46562447/a9d84e24-0681-4ea7-862f-73e3df8caa3e">

This one: CarbonDioxidePeakLevel I would like to have a mean on iphone to reset, so push button would be nice.
How can I create a push button that will be in the place where CO2 is (not somewhere else in Home App)?
Today I am only able to reset peak value by ... restarting the device which is of course not a solution.

Thank you

**Comment by HomeSpan** [comment]
Apple unfortunately did not include any method to do this (they probably were thinking about commercial CO<sub>2</sub> sensors that would be HomeKit compatible, but still include physical buttons on the device, rather than a device that was fully controlled from the Home App itself).

One option to consider is to add a regular Switch Service to the Accessory and use that to trigger a reset of the peak level within HomeSpan.  To simulate a "pushbutton" you can have HomeSpan also reset the state of the switch to "off" after you flip it on.

**Comment by homonto** [comment]
thx, I was "afraid" that was the case ;-)
